tests/ceph_volume/test_ceph_volume_wrong_db_size_compute.py

```diff
@@ -0,0 +1,112 @@
+import json
+
+from cephci.utils.configs import get_cloud_credentials, get_configs
+from cli.cephadm.cephadm import CephAdm
+from cli.cloudproviders import CloudProvider
+from cli.cluster.node import Node
+from cli.cluster.volume import Volume
+from cli.utilities.operations import wait_for_cluster_health
+from cli.utilities.utils import (
+    create_yaml_config,
+    get_disk_devlinks,
+    get_lvm_on_osd_container,
+    get_running_containers,
+)
+
+
+class SpecFileApplyError(Exception):
+    pass
+
+
+def _add_node_volumes(nodes, config, size, count):
+    """Create volume in bulk and attach to node
+    Args:
+        nodes (ceph): nodes on volumes to create
+        config (dict): ceph config
+        size (str): size of volume
+        count (int): number of volume
+    """
+    # Get cloud creadentials
+    cloud_type = config.get("cloud-type")
+    cloud_configs = get_cloud_credentials(cloud_type)
+    cloud = CloudProvider(cloud_type, **cloud_configs)
+
+    nodes = nodes if type(nodes) in (list, tuple) else [nodes]
+    for node in nodes:
+        for i in range(count):
+            volname = f"{node.ceph_nodename}-vol-{size}-{i}"
+            Volume(volname, cloud).create(size=size)
+            Node(node.ceph_nodename, cloud).attach_volume(volname)
+    return True
+
+
+def _identify_disks(node):
+    """
+    Identifies the disks associated with the node
+    Args:
+        node (ceph): Node in which the cmd is executed
+
+    Returns (list): List of available disks
+    """
+
+    disks = []
+    conf = {"format": "json"}
+    available_disks = json.loads(CephAdm(node).ceph.orch.device.ls(**conf))
+    for key in available_disks:
+        if key.get("devices"):
+            for dev in key.get("devices"):
+                disks.append(dev.get("path"))
+    return list(set(disks))
+
+
+def run(ceph_cluster, **kw):
+    """Wrong block_db_size computed when adding OSD"""
+
+    # Get configs
+    get_configs()
+    config = kw.get("config")
+
+    # Get the installer and OSD nodes
+    installer = ceph_cluster.get_nodes(role="installer")[0]
+    osd_nodes = ceph_cluster.get_nodes(role="osd")
+
+    # Get device spec
+    device_spec = config.get("spec", {}).get("spec", {})
```

Comment:
@adityaramteke This will fail as there is no section named `spec` in the suite. This has to be changed to `specs` as per the suite file.